### ISSUE TITLE
Rendering BayesNets when you don't have LaTeX

### ISSUE THREAD

**Issue by tawheeler** [sisl/BayesNets.jl#56]
This is similar to #53, but a focus on a temporary workaround.

We would like BNs to be rendered as text if you do not have LaTeX. The stack is as follows:

- BayesNets: `show(f::IO, a::MIME"image/svg+xml", bn::BayesNet)`
- TikzGraphs: `plot(g::SimpleGraph)`
- TikzPictures: `show(f::IO, ::MIME"image/svg+xml", tp::TikzPicture)`

Our options are:

1. Modify `show` in BayesNets to produce text. This is a little weird since it is in a MIME"image/svg+xml". We could conditionally generate the `show` method but that is intuitive
2. Modify `plot` in TikzGraphs. This is probably a bad idea because it is supposed to generate a  `TikzPicture`
3.  Modify `show` in TikzPictures. It makes most sense to have TikzPictures reason about whether lualatex is installed or not, but when passed a TikzPicture its job is just to render it, not determine whether it is a graph and print it out.

I am leaning towards modifying BayesNets.

**Comment by mykelk** [comment]
Yeah, I think just modify BayesNets. You can do a quick check to see if TikzGraphs fails and then set the following appropriately to output text:
```julia
Base.mimewritable(::MIME"image/svg+xml", bn::BayesNet) = true
Base.mimewritable(::MIME"text/html", dfs::Vector{DataFrame}) = true
```

**Comment by hpoit** [comment]
Tim, I still couldn't get pdf2svg to work on OSX, even after adding it to the path
![image](https://cloud.githubusercontent.com/assets/3711100/22713085/2e90dcfc-ed6e-11e6-9660-e3b64bea82e5.png)

**Comment by mykelk** [comment]
@hpoit Did you restart the console that launched your notebook? Make sure that the command runs in the console before you launch jupyter. If it doesn't then it was not added correctly to your path.

**Comment by hpoit** [comment]
Hi Mykel, I actually had problems with `brew install pdf2svg`
![image](https://cloud.githubusercontent.com/assets/3711100/22714391/68183e7a-ed73-11e6-93e5-29476eba9252.png)

**Comment by tawheeler** [comment]
So a temporary workaround was issued in 21d211353074f69fe5daf679146dd1f04adfa47b

**Comment by hpoit** [comment]
Can we use http://www.graphviz.org/content/world? Once you click on the graph image, you'll be directed to its text file. 

https://github.com/Keno/GraphViz.jl

**Comment by tawheeler** [comment]
GraphViz sounds like exactly what we need. Can you try forking BayesNets and modifying the `show` function to use that and see how it does?

**Comment by mykelk** [comment]
I'm not sure it is seamless for Windows users to install GraphViz.jl because of binary dependencies. There is an open issue on this.

**Comment by tawheeler** [comment]
Do you think that's a show stopper? I figure if there is pressure to get that resolved it will eventually be fixed, and we can leverage the existing work rather than writing something from scratch.

**Comment by mykelk** [comment]
I think there are others without complicated dependencies here: https://github.com/JuliaGraphs

**Comment by hpoit** [comment]
Thanks Mykel. Let me see what I can concoct from JuliaGraphs.

**Comment by tawheeler** [comment]
Temporary fix implemented - rest of problem deferred to #53
